rules/aip0217/return_partial_success_with_unreachable.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0217
+
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var returnPartialSuccessWithUnreachable = &lint.MethodRule{
+	Name: lint.NewRuleName(217, "return-partial-success-with-unreachable"),
+	OnlyIf: func(m *desc.MethodDescriptor) bool {
+		return m.GetInputType().FindFieldByName("return_partial_success") != nil
```

Comment:
dumb question: does the `m.GetInputType()` and `m.GetOutputType()` correspond to the same verb's Req and Resp objects respectively for a given resource?

Reply:
It's simpler than that. In pure protobuf descriptor protos, these are just `string` fields ([here](https://github.com/protobufjs/protobuf.js/blob/0a0cdb6b0c635d86a9bb96a382b4ff73ee848109/google/protobuf/descriptor.proto#L125-L126)) containing the fully qualified name of the types that are declared in the RPC as the request and response messages. This `protoreflect` library we use is a wrapper that fully resolves those to the actual type. So when we invoke `GetInputType` it provides the actual descriptor for the type declared as the request message for the RPC.